src/ComponentConcerns/ValidatesInput.php

```diff
@@ -153,7 +153,12 @@ public function validate($rules = null, $messages = [], $attributes = [])
 
         $this->shortenModelAttributes($data, $rules, $validator);
 
-        $validatedData = $validator->validate();
+        try {
+            $validatedData = $validator->validate();
+        } catch (ValidationException $e) {
+            $this->emit('failed-validation');
+            throw new ValidationException($validator);
```

Comment:
To reduce memory usage, you can simply rethrow the catched exception: `throw $e`.

Reply:
Thanks for your help and patience. Every day is a school day.